apps/examples/fs_test/smartfs_test.c

```diff
@@ -0,0 +1,337 @@
+			close(fd);
+			return ERROR;
+		}
+		total_read += ret;
+	}
+	timediff = get_time() - time;
+	g_total_time += timediff;
+	print_log("read:%dms, ", timediff);
+
+	close(fd);
+
+	return OK;
+}
+
+static int fs_remove(char *filepath, void *buf, int nbytes)
+{
+	int ret;
+	long time;
+	long timediff;
+
+	time = get_time();
+	ret = unlink(filepath);
+	if (ret < 0) {
+		printf("%s errno: %d\n", __FUNCTION__, errno);
+		return ERROR;
+	}
+	timediff = get_time() - time;
+	g_total_time += timediff;
+	print_log("remove:%dms, ", timediff);
+
+	return OK;
+}
+
+static int write_read_test(char *filepath, void *buf, int nbytes, int iteration)
+{
+	int i;
+	int ret;
+
+	for (i = 0; i < iteration; i++) {
+		ret = fs_write(filepath, buf, nbytes);
+		if (ret < 0) {
+			return ERROR;
+		}
+
+		ret = fs_read(filepath, buf, nbytes);
+		if (ret < 0) {
+			return ERROR;
+		}
+	}
+
+	return OK;
+}
+
+static int write_test(char *filepath, void *buf, int nbytes, int iteration)
+{
+	int i;
+	int ret; 
+
+	for (i = 0; i < iteration; i++) {
+		ret = fs_write(filepath, buf, nbytes);
+		if (ret < 0) {
+			return ERROR;
+		}
+	}
+
+	return OK;
+}
+
+static int read_test(char *filepath, void *buf, int nbytes, int iteration)
+{
+	int i;
+	int ret;
+
+	ret = fs_write(filepath, buf, nbytes);
+	if (ret < 0) {
+		return ERROR;
+	}
+
+	for (i = 0; i < iteration; i++) {
+		ret = fs_read(filepath, buf, nbytes);
+		if (ret < 0) {
+			return ERROR;
+		}
+	}
+
+	return OK;
+}
+
+static int write_remove_test(char *filepath, void *buf, int nbytes, int iteration)
+{
+	int i;
+	int ret;
+
+	for (i = 0; i < iteration; i++) {
+		ret = fs_write(filepath, buf, nbytes);
+		if (ret < 0) {
+			return ERROR;
+		}
+
+		ret = fs_remove(filepath, buf, nbytes);
+		if (ret < 0) {
+			return ERROR;
+		}
+	}
+
+	return OK;
+}
+
+void run_fs_test(int operation, char *filepath, int size, int iteration)
+{
+	int ret;
+	char *buf;
+
+	buf = (char *)malloc(size);
+	if (!buf) {
+		printf("%s: Malloc failed, %d\n", __FUNCTION__, size);
+		return;
+	}
+
+	memset(buf, 0, size);
+
+	switch (operation) {
+	case TEST_OPT_WRITEREAD:
+		ret = write_read_test(filepath, buf, size, iteration);
+		break;
+	case TEST_OPT_READ:
+		ret = read_test(filepath, buf, size, iteration);
+		break;
+	case TEST_OPT_WRITE:
+		ret = write_test(filepath, buf, size, iteration);
+		break;
+	case TEST_OPT_WRITEREMOVE:
+		ret = write_remove_test(filepath, buf, size, iteration);
+		break;
+
+	default:
+		break;
+	}
+	free(buf);
+
+	if (ret == OK) {
+		printf("run_time: %d ms\n", g_total_time);
+	}
+}
+
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+int smartfs_test_main(int argc, char *argv[])
+#endif
+{
+	int size;
+	int operation;
+	int iteration;
+	char *filepath;
+
+	printf("FS R/W Performance Test!\n");
+
+	if (argc <= 5) {
+		printf("%s", USAGE);
+		return 0;
+	}
+
+	operation = atoi(argv[1]);
+	filepath = argv[2];
+	size = atoi(argv[3]);
+	iteration = atoi(argv[4]);
+	g_printopt = argv[5][0];
```

Comment:
You checked the `argc` with smaller than equal to 5, but you used the index as 5 in `argv[][]`.
It is not correct.
If `argc` is really smaller than equal to 5, then we should not access the `argv` with index 5.
If not, please change the `argc` checking logic.